src/coreclr/vm/jitinterface.cpp

```diff
@@ -8914,16 +8913,19 @@ CORINFO_CLASS_HANDLE CEEInfo::getDefaultEqualityComparerClassHelper(CORINFO_CLAS
         MODE_PREEMPTIVE;
     } CONTRACTL_END;
 
-    // Mirrors the logic in BCL's CompareHelpers.CreateDefaultEqualityComparer
-    // And in compile.cpp's SpecializeEqualityComparer
+    // Mirrors the logic in BCL's CompareHelpers.CreateDefaultEqualityComparer.
     TypeHandle elemTypeHnd(elemType);
 
-    // Mirrors the logic in BCL's CompareHelpers.CreateDefaultComparer
-    // And in compile.cpp's SpecializeComparer
-    //
-    // We need to find the appropriate instantiation
+    // We need to find the appropriate instantiation.
     Instantiation inst(&elemTypeHnd, 1);
 
+    // string
+    if (elemTypeHnd.IsString())
+    {
+        TypeHandle resultTh = ((TypeHandle)CoreLibBinder::GetClass(CLASS__STRING_EQUALITYCOMPARER)).Instantiate(inst);
+        return CORINFO_CLASS_HANDLE(resultTh.GetMethodTable());
+    }
```

Comment:
Is there a runtime test that makes sure this is working properly in shared generics?

Reply:
StringEqualityComparer is not a generic type. I do not see any new interesting cases for shared generics.

Reply:
> StringEqualityComparer is not a generic type. I do not see any new interesting cases for shared generics.

I mean checking if `EqualityComparer<string>.Default` properly returns `StringEqualityComparer` in shared generic code.